wordpress-mobile/WordPress-iOS
# Abstract post list to Swift.

### Description:

Migrates AbstractPostListViewController to Swift. Since porting just a part of the code is not feasible, I'm describing below how this'll be taken care of.

### Scope:

**Changes covered by the scope of this PR:**
- Porting the code from ObjC to Swift.
- Making sure there are no regressions in the porting process.

**Changes not covered by the scope of this PR:**
- Fixing bugs in the original code.
- Improving the original code.
- Adding documentation to the original code.

I realize these are important when adding new code, but in a porting process they can become detrimental due to a loss of focus.

You can still point out possible improvements, and if they make sense I'll make sure to open new tickets.

### Code Review:

I've split the review in sections (the `#pragma mark` sections from the original ObjC file), so that the reviewers can focus on smaller parts of the whole thing.

Please review @jleandroperez 
- [x] Header File
- [x] Constants & Properties
- [x] Lifecycle Methods
- [x] Multitasking support
- [x] Synching
- [x] Sync Helper Delegate Methods

Please review @SergioEstevao  
- [x] TableView Handler Delegate Methods
- [x] Table View Handling
- [x] Actions (there are two "Actions" sections in the original file... this is the second one).
- [x] Post actions
- [x] Search Related
- [x] Filter related
- [x] Search Controller Delegate Methods

Please review @kurzee :
- [x] Changes to `PageListViewController.swift`.
- [x] Changes to `PostListViewController.swift`.

Notes: the reason behind the changes you are going to review:
- Class no longer needs to be `@objc` since the base class is now written in Swift.
- Some methods which were being overriden now require the `override` keyword.  Adding such keyword when the parent class was written in ObjC was producing an error, but it's not required to compile.
- Some members of the base class (`AbstractPostListViewController`) which were optionals are no longer optionals.

### Testing:

Please make sure to run these tests both with WP.com and self-hosted, and both for pages and posts.

Tests for @jleandroperez :

1. Make sure you open a blog without posts/pages.
2. Make sure that even if you select any filter from the top, no posts/pages are shown on screen.
3. Make sure that each section is showing an appropriate message at the center of the screen (you'll know when you read them!).

1. Make sure you open a blog without posts/pages.
2. Tap the search icon.
3. Type in any word.
4. Make sure the text at the center of the screen shows an appropriate message for the filter you've selected.

Tests for @SergioEstevao :

Same as test 2 but with results.  Make sure the results are shown properly.

1. Add pages through the blog's web page.
2. Make sure at least 1 of them is published, 1 draft, 1 scheduled and 1 trashed.
3. Check in the app to make sure the posts/pages are in the appropriate section.

Tests for @kurzee :

1. Same as test 4 but once you have the posts/pages...
2. Move them around or delete them using WPiOS, and make sure the changes are properly reflected in the web interface.


- [x] Test 6:
1. Create a new post/page in WPiOS for each one of these: published, draft, scheduled.
2. Make sure they show up properly and that they're uploaded to your blog.

And any other test you can think of in that screen. :)
